modin/experimental/xgboost/test/test_dmatrix.py

```diff
@@ -0,0 +1,166 @@
+            ), "Got Modin Exception type {}, but xgboost Exception type {} was expected".format(
+                type(mxgb_exception.value), type(xgb_exception)
+            )
+        else:
+            assert mxgb_exception.type is ValueError
+    else:
+        md_dm = mxgb.DMatrix(modin_data, label=modin_label, **kwargs)
+        assert md_dm.num_row() == dm.num_row()
+        assert md_dm.num_col() == dm.num_col()
+        assert md_dm.feature_names == dm.feature_names
+        assert md_dm.feature_types == dm.feature_types
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.randn(5, 5),
+        np.array([[1, 2], [3, 4]]),
+        np.array([["a", "b"], ["c", "d"]]),
+        [[1, 2], [3, 4]],
+        [["a", "b"], ["c", "d"]],
+    ],
+)
+@pytest.mark.parametrize(
+    "feature_names",
+    [
+        list("abcdef"),
+        ["a", "b", "c", "d", "d"],
+        ["a", "b", "c", "d", "e<1"],
+        list("abcde"),
+    ],
+)
+@pytest.mark.parametrize(
+    "feature_types",
+    [None, "q", list("qiqiq")],
+)
+def test_dmatrix_feature_names_and_feature_types(data, feature_names, feature_types):
+    check_dmatrix(data, feature_names=feature_names, feature_types=feature_types)
+
+
+@pytest.mark.parametrize(
+    "feature_names",
+    [
+        ["Feature1", "Feature2", "Feature3", "Feature4", "Feature5"],
+        [u"??1", u"??2", u"??3", u"??4", u"??5"],
+    ],
+)
+def test_feature_names(feature_names):
+    data = np.random.randn(100, 5)
+    label = np.array([0, 1] * 50)
+
+    check_dmatrix(
+        data,
+        label,
+        feature_names=feature_names,
+    )
+
+    dm = xgb.DMatrix(data, label=label, feature_names=feature_names)
+    md_dm = mxgb.DMatrix(
+        pd.DataFrame(data), label=pd.Series(label), feature_names=feature_names
+    )
+
+    params = {
+        "objective": "multi:softprob",
+        "eval_metric": "mlogloss",
+        "eta": 0.3,
+        "num_class": 3,
+    }
+
+    bst = xgb.train(params, dm, num_boost_round=10)
+    md_bst = mxgb.train(params, md_dm, num_boost_round=10)
+
+    predictions = bst.predict(dm)
+    modin_predictions = md_bst.predict(md_dm)
+
+    preds = np.asarray([np.argmax(line) for line in predictions])
+    md_preds = np.asarray([np.argmax(line) for line in modin_predictions.to_numpy()])
+
+    val = accuracy_score(label, preds)
+    md_val = accuracy_score(label, md_preds)
+
+    np.testing.assert_allclose(val, md_val, atol=0.02, rtol=0.01)
+
+    dummy = np.random.randn(5, 5)
+    dm = xgb.DMatrix(dummy, feature_names=feature_names)
+    md_dm = mxgb.DMatrix(pd.DataFrame(dummy), feature_names=feature_names)
+    predictions = bst.predict(dm)
+    modin_predictions = md_bst.predict(md_dm)
+
+    preds = np.asarray([np.argmax(line) for line in predictions])
+    md_preds = np.asarray([np.argmax(line) for line in modin_predictions.to_numpy()])
+
+    assert preds.all() == md_preds.all()
+
+    # different feature name must raises error
```

Comment:
Why doesn't this case depend on the original parameter `feature_names` of the test?

Reply:
```suggestion
    # different feature names must raises error
```

Reply:
Where checks values after work functions, when ``feature_names`` using in build of dmatrix